tests/integration-tests/big-decimal/src/mapping.ts

```diff
@@ -6,21 +6,27 @@ export function handleTrigger(event: Trigger): void {
   // big_decimal exponent will be: -35 - 6109 = -6144.
   // Minmum exponent is: -6143.
   // So 1 digit will be removed, the 8, and the 6 will be rounded to 7.
-  let big_decimal = BigDecimal.fromString(
-    "0.99999999999999999999999999999999968"
-  );
+  let small = BigDecimal.fromString("0.99999999999999999999999999999999968");
 
-  big_decimal.exp -= BigInt.fromI32(6109);
+  small.exp -= BigInt.fromI32(6109);
 
   // Round-trip through the node so it truncates.
-  big_decimal = big_decimal * new BigDecimal(BigInt.fromI32(1));
+  small = small * new BigDecimal(BigInt.fromI32(1));
 
-  assert(big_decimal.exp == BigInt.fromI32(-6143), "wrong exp");
+  assert(small.exp == BigInt.fromI32(-6143), "wrong exp");
 
   // This has 33 nines and the 7 which was rounded from 6.
   assert(
-    big_decimal.digits ==
+    small.digits ==
       BigDecimal.fromString("9999999999999999999999999999999997").digits,
-    "fug " + big_decimal.digits.toString()
+    "wrong digits " + small.digits.toString()
   );
+
+  // This has 35 nines, but we round to 34 decimal digits.
+  let big = BigDecimal.fromString("99999999999999999999999999999999999");
+
+  // This has 35 zeros.
+  let rounded = BigDecimal.fromString("100000000000000000000000000000000000");
```

Comment:
This test case doesn't prove as much as it could. It would be better to use `8` instead of `9` to demonstrate exactly where the rounding occurs.

Reply:
I've added that test case.

Reply:
I've added that test case.